app/src/main/java/com/mk/tasky/MainViewModel.kt

```diff
@@ -32,4 +39,13 @@ class MainViewModel @Inject constructor(
             }
         }
     }
+
+    fun onLogout() {
+        viewModelScope.launch {
+            preferences.deleteUser()
+            withContext(defaultDispatcher) {
+                database.clearAllTables()
```

Comment:
Why the default dispatcher for an IO thing?

Reply:
Will rename variable. nice catch!